tracing-subscriber/src/fmt/format/json.rs

```diff
@@ -51,8 +61,20 @@ where
             serializer.serialize_entry("timestamp", &timestamp)?;
             serializer.serialize_entry("level", &meta.level().as_serde())?;
 
-            ctx.with_current(|(_, span)| serializer.serialize_entry("span", &span))
-                .unwrap_or(Ok(()))?;
+            let id = ctx.ctx.current_span();
+            let id = id.id();
+            if let Some(id) = id {
+                if let Some(span) = ctx.ctx.span(id) {
+                    let ext = span.extensions();
+                    let data = ext
+                        .get::<FormattedFields<N>>()
+                        .expect("Unable to find FormattedFields in extensions; this is a bug");
+                    let data = &data.fmt_fields;
+                    let mut fields: Value = serde_json::from_str(&data)?;
```

Comment:
Why does this now need to _parse_ JSON where it didn't before?

Reply:
I think that's a solution for #391

Reply:
This resolves #391. We should probably rework this to use a `serde_json::Value` in a JSON-specific layer, but I'd rather have a uglier fix _now_ rather than shipping broken JSON.

Reply:
Okay, so I guess we're just including the fix in this PR. Seems fine. Can we add a `// TODO` here though to remind ourselves about fixing that later though?

Reply:
Sure!

Reply:
Added a comment.